packages/dds/tree/src/test/shared-tree/sharedTreeFuzzTests.ts

```diff
@@ -0,0 +1,287 @@
+        },
+        initialState,
+        saveInfo,
+    );
+
+    await finalState.testTreeProvider.ensureSynchronized();
+    validateTree(provider.trees[0], [initialTreeState]);
+
+    if (validateAnchor) {
+        const expectedPath: UpPath = {
+            parent: {
+                parent: undefined,
+                parentIndex: 0,
+                parentField: rootFieldKeySymbol,
+            },
+            parentField: brand("foo"),
+            parentIndex: 1,
+        };
+        const anchorPath = tree.locate(firstAnchor);
+        assert(compareUpPaths(expectedPath, anchorPath));
+    }
+    return finalState;
+}
+
+function applyFuzzChange(tree: ISharedTree, contents: FuzzChange, transactionResult:TransactionResult): void {
+    const index = contents.index;
+    const nodeField = contents.field;
+    switch (contents.fuzzType) {
+        case "insert":
+            if (index !== undefined && nodeField !== undefined) {
+                tree.runTransaction((forest, editor) => {
+                    const field = editor.sequenceField(contents.parent, nodeField);
+                    field.insert(
+                        index,
+                        singleTextCursor({ type: brand("Test"), value: contents.value }),
+                    );
+                    return transactionResult;
+                });
+            }
+            break;
+        case "delete":
+            if (index !== undefined && nodeField !== undefined) {
+                const parent = contents.parent?.parent;
+                const delField = contents.parent?.parentField;
+                const parentIndex = contents.parent?.parentIndex;
+                if (delField !== undefined && parent !== undefined && parentIndex !== undefined) {
+                    tree.runTransaction((forest, editor) => {
+                        const field = editor.sequenceField(parent, delField);
+                        field.delete(parentIndex, 1);
+                        return transactionResult;
+                    });
+                }
+            }
+            break;
+        case "setPayload":
+            if (index !== undefined && nodeField !== undefined) {
+                const path = contents.parent;
+                if (path !== undefined) {
+                    tree.runTransaction((forest, editor) => {
+                        editor.setValue(path, contents.value);
+                        return transactionResult;
+                    });
+                }
+            }
+            break;
+        default:
+            fail("Invalid edit.");
+    }
+}
+
+enum FuzzTestType {
+    Basic = 0,
+    Abort = 1,
+    AnchorStability = 2,
+}
+
+export function runSharedTreeFuzzTests(title: string): void {
+    const random = makeRandom(0)
+    const testBatchSize = 3;
+    describeFuzz(title, ({ testCount }) => {
+        function runFuzzTest (
+            opGenerator: () => AsyncGenerator<Operation, FuzzTestState>,
+            fuzzTestType: FuzzTestType,
```

Comment:
If someone wants to run the fuzz tests with a new "perform" function, they would need to update the enum declaration, call `runSharedTreeFuzzTests` with the new enum value, and update the switch statement to add a case for it. The first two things are not so bad, but requiring users of `runSharedTreeFuzzTests` to update its code when wanting to use it with different parameters is far from ideal.

Instead of using an enum here, you can take a delegate (that takes as input the AsyncGenerator and the seed). The caller of `runSharedTreeFuzzTests` can then pass in the delegate they want. This will also get rid of the code duplication between the switch branches.

Reply:
`performFuzzActionsAbort` function has an extra optional parameter `validateAnchor`, compared to `performFuzzActions`. Since the delegate only takes in seed and async generator, should these two functions be combined? Or would it be better to remove the `validateAnchor` boolean, and just keep the anchor validation always on?

Reply:
I was thinking you could make a delegate like this `(generator, seed) => performFuzzActionsAbort(generator, seed, true)`, but now that I think about it more, I don't think we really need to have separate tests for the abort (one that checks the anchors are stable and one that doesn't). Instead we can just check the anchors all the time. This means the boolean parameter can go away.